events/views.py

```diff
@@ -123,9 +123,20 @@ def updateEvent(request, event_id):
                 event_location_id = int(event_location_id)
                 if event_location_id <= 0:
                     errors["event_location_id"] = "Event location must be selected."
+                else:
+                    location_object = Location.objects.get(id=event_location_id)
+                    if Event.objects.filter(
+                        event_name=event_name,
+                        event_location=location_object,
+                        start_time=start_time,
+                        end_time=end_time,
+                        is_active=True,
+                    ).exists():
+                        errors[
+                            "similar_event_error"
+                        ] = "An event with these details already exists."
             except ValueError:
```

Comment:
We need to add the check here to ensure that only the creator of the event can update the event